pkg/config/expand.go

```diff
@@ -513,20 +512,46 @@ type varContext struct {
 	blueprint  Blueprint
 }
 
+type reference interface {
+	validate(Blueprint) error
+	isIntergroup() bool
+	String() string
+	getFromModuleID() string
+	getToModuleID() string
```

Comment:
I think it is idiomatic for Go to drop the `get` from the getters: https://go.dev/doc/effective_go#Getters 

There is a lack of clarity there because it mentions the lower/upper case being sufficient to tell them apart. You are making these fields private so that difference is not present. Thoughts? It also seems odd to me that all of the underlying data is public but the interface methods are private, should it not be the other way around?

Reply:
Modified on PR branch.

Reply:
I believe this is done, close if you agree.